src/ec2/prompter.ts

```diff
@@ -3,28 +3,36 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { createQuickPick, QuickPickPrompter } from "../shared/ui/pickerPrompter"
-import { DataQuickPickItem } from "../shared/ui/pickerPrompter"
-import { AsyncCollection } from "../shared/utilities/asyncCollection"
-import { isValidResponse } from '../shared/wizards/wizard'
+import { RegionSubmenu, RegionSubmenuResponse } from '../shared/ui/common/regionSubmenu'
+import { getInstanceIdsFromRegion } from "./utils";
+import { DataQuickPickItem } from "../shared/ui/pickerPrompter";
 
-function asQuickpickItem(stringItem: string): DataQuickPickItem<string>[] 
-{
-    return [{
-        label: stringItem, 
-        data: stringItem
-    }]
+type EC2InstanceId = string
+
+type EC2Selection = {
```

Comment:
nit: use `interface` for object-like shapes

It's _mostly_ a style thing.

Reply:
So should `type` be reserved for enum-like shapes. Ex. `type currency = 'USD' | 'EUR'`. Are there other cases where `type` is preferred?

Reply:
> So should type be reserved for enum-like shapes. 

I pretty much only use `type` when I can't use `interface`. Which is actually pretty frequent with "advanced" types.